pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ControllerFilePathProvider.java

```diff
@@ -126,14 +128,38 @@ public URI getDataDirURI() {
   }
 
   public File getFileUploadTempDir() {
+    if (!Files.exists(_fileUploadTempDir.toPath())) {
```

Comment:
Extract this logic into a method to avoid replicating it.

Reply:
+1, create a method `ensureUploadTempDirExists`

Reply:
done